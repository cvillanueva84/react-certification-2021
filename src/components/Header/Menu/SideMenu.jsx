import React, { useRef } from 'react';
import { Link } from 'react-router-dom';
import {
  HamburgerMenu,
  Menu,
  MenuButton,
  MenuCover,
  SpanHamburguer,
} from './Menu.styled';

const SideMenu = () => {
  const ref = useRef(null);

  const hidenMenu = () => {
    ref.current.checked = false;
  };

  return (
    <HamburgerMenu>
      <MenuButton ref={ref} type="checkbox" data-testid="hamburguerMenu" />
      <MenuCover onClick={hidenMenu} />
      <SpanHamburguer />
      <SpanHamburguer />
      <SpanHamburguer />

      <Menu>
        <Link to="/" onClick={hidenMenu}>
          <li>Home</li>
        </Link>
        <Link to="/" onClick={hidenMenu}>
          <li>Home</li>
        </Link>
      </Menu>
    </HamburgerMenu>
  );
};

export default SideMenu;
