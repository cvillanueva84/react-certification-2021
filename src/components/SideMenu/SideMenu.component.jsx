import React, { useContext } from 'react';

import { AiFillHome, AiFillStar } from 'react-icons/ai';
import { FaSun, FaMoon } from 'react-icons/fa';
import ThemeContext from '../Context/ThemeContext';
import LinkVideo from '../Link.element';
import { SideMenuContainer, SideMenuItem } from './SideMenu.elements';

import { useAuth } from '../../providers/Auth';

const PrivateLink = () => {
  const { authenticated } = useAuth();

  return authenticated ? (
    <LinkVideo to="/favorites">
      <SideMenuItem Icon={AiFillStar} text="Liked" />
    </LinkVideo>
  ) : (
    <LinkVideo to="/login">
      <SideMenuItem Icon={AiFillStar} text="Liked" />
    </LinkVideo>
  );
};

function SideMenu() {
  const { theme, setTheme } = useContext(ThemeContext);

  const icon = theme === 'light' ? FaMoon : FaSun;
  const text = theme === 'light' ? 'Dark Mode' : 'Light Mode';

  const handleTheme = () => setTheme(theme === 'light' ? 'dark' : 'light');

  return (
    <>
      <SideMenuContainer>
        <LinkVideo to="/">
          <SideMenuItem Icon={AiFillHome} text="Home" />
        </LinkVideo>
        <PrivateLink />
        <SideMenuItem Icon={icon} text={text} fun={handleTheme} />
      </SideMenuContainer>
    </>
  );
}

export default SideMenu;
