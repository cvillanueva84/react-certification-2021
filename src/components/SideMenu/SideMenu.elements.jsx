import React from 'react';
import styled from 'styled-components';

export const SideMenuContainer = styled.ul`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  position: fixed;
  top: ${(props) => props.theme.navbar_height};
  left: 0;
  height: calc(100vh - ${(props) => props.theme.navbar_height});
  width: ${(props) => props.theme.sidemenu_width};
  background: ${(props) => props.theme.background_color};
`;

const SideMenuOption = styled.li`
  color: ${(props) => props.theme.icon_color};
  text-decoration: bold;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0.5rem;
  user-select: none;
  font-weight: bold;
  width: 100%;

  &:hover {
    background: ${({ theme }) => theme.scrollbar_thumb_hover_color};
  }
`;

export function SideMenuItem({ Icon, text }) {
  return (
    <SideMenuOption>
      <Icon size="2rem" />
      {text}
    </SideMenuOption>
  );
}
