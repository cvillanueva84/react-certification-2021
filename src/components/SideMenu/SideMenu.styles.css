.side-menu-container {
  background-color: #c2c3ff;
  color: black;
  height: 0;
  left: 0;
  min-width: 250px;
  overflow: hidden;
  position: absolute;
  top: 3.5em;
  width: 16%;
  transition: height 0.25s ease-in;
  -webkit-transition: height 0.25s ease-in;
}

.side-menu-container.active {
  -webkit-box-shadow: 0px 10px 13px -7px #000000, 5px 9px 12px 5px rgba(0, 0, 0, 0.35);
  box-shadow: 0px 10px 13px -7px #000000, 5px 9px 12px 5px rgba(0, 0, 0, 0.35);
  height: calc(100% - 3.5em);
  transition: height 0.25s ease-in;
  -webkit-transition: height 0.25s ease-in;
}
