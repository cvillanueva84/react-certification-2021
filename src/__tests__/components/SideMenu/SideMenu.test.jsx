import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import SideMenu from '../../../components/SideMenu/SideMenu.component';

let tree;

beforeEach(() => {
  tree = renderer.create(<SideMenu />).toJSON();
});

describe('SideMenu', () => {
  it('Renders SideMenu', () => {
    expect(tree).toMatchSnapshot();
  });

  it('Displays side menu content', () => {
    render(<SideMenu toggleSideMenu={null} />);
    const sideMenuComponent = screen.getByText('This is the side menu :)');
    expect(sideMenuComponent).toBeInTheDocument();
  });
});
