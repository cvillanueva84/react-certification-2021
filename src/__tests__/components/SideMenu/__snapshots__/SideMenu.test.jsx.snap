// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu Renders SideMenu 1`] = `
<div
  className="sc-bdnxRM gvAXaq"
>
  <div
    className="sc-dlnjwi jKImPw"
    onClick={[Function]}
  >
    <span
      className="sc-hKFxyN fZULHG"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </span>
  </div>
  <div
    className="sc-eCApnc jjUpuo"
  />
  <div
    className="sc-gtsrHT dkYNPb"
  >
    <span>
      This is the side menu :)
    </span>
  </div>
</div>
`;
